playground/negotiator-authenticator.js

```diff
@@ -0,0 +1,154 @@
+
+// Mock value
+const currentUser = {
+    ownerAddress: 2
+};
+
+let tokenMock = [
+    { token: { ticketId: 42, ticketClass: "VIP", conferenceId: 1 }, ownerAddress: null },
+    { token: { ticketId: 32, ticketClass: "STANDARD", conferenceId: 1 }, ownerAddress: 2 },
+    { token: { ticketId: 15, ticketClass: "VIP", conferenceId: 1 }, ownerAddress: 2 },
+];
+
+// Negotiator Module
+const Negotiator = {
+    getTokenInstances: function async() { return tokenMock }
+};
+
+// Authenticator Module
+const Authenticator = {
+    authenticateAddress: async function (ownerAddress) {
+        return true;
+    },
+    assetOwnerAddress: async function () {
+        tokenMock[0].token.ownerAddress = 2;
+        return tokenMock[0].token.ownerAddress;
+    },
+    authenticate: async function () {
+        return "Authenticated"
+    }
+};
+
+// Helper for success and failed promises
+// negotiatorAuthenticator code 
+const asyncHelper = async (promise) => {
+    try {
+        const data = await promise;
+        return [data, undefined];
+    } catch (error) {
+        return [undefined, error];
+    }
+}
+
+const handlerSubmit = (event) => {
+    event.preventDefault();
+    Authenticator.assetOwnerAddress(event.currentTarget[0].value).then(function (result) {
+        // sequence of steps to trigger...
+        // authenticateAddress(ownerAddress)
+        // authenticate(chosenTicket)
+        // Close Modal, provide permission to VIP Room
+        document.getElementById("modal").style.display = 'none';
+        alert('user can enter the VIP room');
+    }, function (error) {
+        // handle error
+    });
+}
+
+const negotiatorAuthenticator = {};
+
+negotiatorAuthenticator.init = async () => {
```

Comment:
TODO - I will look at the naming used here and tidy it. Where if I am correct, we should only have Negotiator and Authenticator modules for the front end API.